drivers/adc/ad7616/ad7616.c

```diff
@@ -506,6 +830,95 @@ int32_t ad7616_core_setup(struct ad7616_dev *dev)
 
 	return 0;
 }
+#endif
+
+/**
+ * @brief Setup the sequencer layers
+ * @param dev The device structure
+ * @param layers Arrays of layers
+ * @param layers_nb Nubers of items in the layers array.
+ * @param burst 1 if needed, 0 otherwise
+ * @return 0 on success, an error value otherwise
+ */
+int32_t ad7616_setup_sequencer(struct ad7616_dev *dev,
+			       struct ad7616_sequencer_layer *layers,
+			       uint32_t layers_nb, uint8_t burst)
+{
+	int ret;
+	for (uint32_t i = 0; i < layers_nb; i++) {
+		uint16_t data = 0;
+
+		// Last one ?
+		if (i == layers_nb - 1)
+			data |= AD7616_SSREN;
+		data |= (layers[i].ch_b - AD7616_VB0) << 4;
+		data |= layers[i].ch_a;
+
+		ret = ad7616_write(dev, AD7616_REG_SEQUENCER_STACK(i), data);
+		if (ret != 0)
+			return ret;
+	}
+
+	if (burst)
+		ret = ad7616_write_mask(dev, AD7616_REG_CONFIG, AD7616_BURSTEN,
+					AD7616_BURSTEN);
+
+	// Enable channel sequencer
+	ret |= ad7616_write_mask(dev, AD7616_REG_CONFIG, AD7616_SEQEN,
```

Comment:
lets avoid doing `red |=` as it really carries no meaning

Reply:
*comment applies throughout the whole file

Reply:
```suggestion
	ret = ret || ad7616_write_mask(dev, AD7616_REG_CONFIG, AD7616_SEQEN,
```

Reply:
I think the best way to deal with theses is with early returns